sdv/metadata/multi_table.py

```diff
@@ -427,6 +432,92 @@ def add_constraint(self, table_name, constraint_name, **kwargs):
         table = self._tables.get(table_name)
         table.add_constraint(constraint_name, **kwargs)
 
+    def _validate_single_table(self, errors):
+        for table_name, table in self._tables.items():
+            try:
+                table.validate()
+            except Exception as error:
+                errors.append('\n')
+                tittle = f'Table: {table_name}'
+                error = str(error).replace(
+                    'The following errors were found in the metadata:\n', tittle)
+                errors.append(error)
+
+    def _validate_disjoined_tables(self, parent_map, child_map):
+        nodes = list(self._tables.keys())
+        queue = [nodes[0]]
+        connected = {table_name: False for table_name in nodes}
+
+        while queue:
+            node = queue.pop()
+            connected[node] = True
+            for child in list(child_map[node]) + list(parent_map[node]):
+                if not connected[child] and child not in queue:
+                    queue.append(child)
+
+        if not all(connected.values()):
+            not_connected_tables = set()
+            for table, value in connected.items():
+                if not value:
+                    not_connected_tables.add(table)
+
+            if len(not_connected_tables) > 1:
+                table_msg = (
+                    f'Tables {not_connected_tables} are not connected to any of the other tables.'
+                )
+            else:
+                table_msg = (
+                    f'Table {not_connected_tables} is not connected to any of the other tables.'
+                )
+
+            raise ValueError(f'The relationships in the dataset are disjointed. {table_msg}')
+
+    def validate(self):
+        """Validate the metadata.
+
+        Raises:
+            - ``InvalidMetadataError`` if the metadata is invalid.
+        """
+        errors = []
+        self._validate_single_table(errors)
+        for relation in self._relationships:
+            try:
+                self._validate_relationship(**relation)
+            except Exception as error:
+                if '\nRelationships:' not in errors:
+                    errors.append('\nRelationships:')
+
+                errors.append(error)
+
+        parent_map = defaultdict(set)
+        child_map = defaultdict(set)
+        for relation in self._relationships:
+            parent_name = relation['parent_table_name']
+            child_name = relation['child_table_name']
+            parent_map[child_name].add(parent_name)
+            child_map[parent_name].add(child_name)
```

Comment:
can we pull this into separate a method? I think it would be useful to have a `_get_child_map` and `_get_parent_map` method. It's also used in `add_relationship`